src/ios/netcdf_variable.F90

```diff
@@ -0,0 +1,557 @@
+    type(string_t) :: musica_name_
+    !> Units for variable in file data
+    type(string_t) :: units_
+    !> Variable id in file
+    integer(kind=musica_ik) :: id_ = -1
+    !> Variable dimensions
+    integer(kind=musica_ik) :: dimensions_(1) = (/ 0 /)
+    !> Converter to MUSICA units
+    type(convert_t) :: converter_
+    !> Scaling factor
+    real(kind=musica_dk) :: scale_factor_ = 1.0_musica_dk
+    !> Offset (applied to file data after scaling and before unit conversion)
+    real(kind=musica_dk) :: offset_ = 0.0_musica_dk
+    !> Shift (applied after unit conversion)
+    real(kind=musica_dk) :: shift_ = 0.0_musica_dk
+  contains
+    !> Returns the NetCDF variable id
+    procedure :: id
+    !> Returns the name of the variable
+    procedure :: name => variable_name
+    !> Returns the MUSICA name for the variable
+    procedure :: musica_name
+    !> Gets the number of entries for the variable in the temporal dimension
+    procedure :: time_dimension_size
+    !> Gets a sub-set of the variable data for a specified index range
+    !!
+    !! Data are returned after applying conversions set up during
+    !! initialization.
+    !!
+    procedure :: get_data
+    !> Prints the properties of the variable
+    procedure :: print => do_print
+    !> Sets the NetCDF <-> MUSICA matching criteria
+    procedure, private :: set_matching_criteria
+    !> Does standard NetCDF -> MUSICA name conversions
+    procedure, private :: do_standard_name_conversions
+    !> Attempts to match to a MUSICA domain state variable
+    procedure, private :: do_match
+    !> Sets any specified data adjustments
+    procedure, private :: set_adjustments
+  end type netcdf_variable_t
+
+  !> Constructor
+  interface netcdf_variable_t
+    module procedure :: constructor_name, constructor_id
+  end interface netcdf_variable_t
+
+  !> Pointer to netcdf_variable_t objects
+  type :: netcdf_variable_ptr
+    type(netcdf_variable_t), pointer :: val_
+  end type netcdf_variable_ptr
+
+contains
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Creates a netcdf_variable_t object for an existing NetCDF variable by name
+  !!
+  !! If no matching state variable is found in the MUSICA domain, a null
+  !! pointer is returned.
+  !!
+  function constructor_name( domain, file, variable_name, config )            &
+      result( new_obj )
+
+    use musica_config,                 only : config_t
+    use musica_domain,                 only : domain_t
+    use musica_netcdf_file,            only : netcdf_file_t
+
+    !> New NetCDF variable
+    type(netcdf_variable_t), pointer :: new_obj
+    !> MUSICA domain
+    class(domain_t), intent(inout) :: domain
+    !> NetCDF file
+    class(netcdf_file_t), intent(inout) :: file
+    !> Variable name
+    character(len=*), intent(in) :: variable_name
+    !> Configuration describing how to match to MUSICA variables
+    !!
+    !! If omitted, standard matching is applied
```

Comment:
Is standard matching anything other than matching based on name?

Reply:
yes, the matching is just based on the name in the input file. If the file uses the standard MusicBox names, they will be automatically matched.